src/app/util/ember-compatibility-functions.cpp

```diff
@@ -178,7 +181,27 @@ void ResetEmberAfObjects()
 namespace {
 // Common buffer for ReadSingleClusterData & WriteSingleClusterData
 uint8_t attributeData[kAttributeReadBufferSize];
-} // namespace
+
+template <typename T>
+CHIP_ERROR attributeBufferToNumericTlvData(TLV::TLVWriter & writer, bool isNullable)
+{
+    typename NumericAttributeTraits<T>::StorageType value;
+    memcpy(&value, attributeData, sizeof(value));
```

Comment:
are there any concerns here with endianess of the underlying platform? Can we somehow validate that this works on both big and little endian machines?

Reply:
Note that this is preserving existing logic....  But in general, we store things in the in-memory attribute store in native endianness, so yes, this is correct.

Validation is tough, but again this PR is not changing the behavior here.